### PR TITLE
fix: warn in case of unexpected terramate block

### PR DESCRIPTION
**Pull request by i4ki**
```
$ cat stack-a/terramate.tm.hcl
stack {
	name = "awesome-stack-a"
}

terramate {

}
$ terramate list
2022-08-04T02:17:37+01:00 WRN  error="/home/i4k/src/mineiros/tm-example/stack-a/terramate.tm.hcl:5,1-10: `terramate` block is only allowed at the project root directory" action=checkConflicts()
stack-a
stack-b
stack-c
```

### CODE REVIEW

File: hcl/hcl.go [hunk 1]
```diff
@@ -190,6 +191,7 @@ type TerramateParser struct {
 	// parsedFiles stores a map of all parsed files
 	parsedFiles map[string]parsedFile
 
+	strict bool
```
Comment (katcipis) [inline hcl/hcl.go:194]: JS feelings :laughing:

File: hcl/hcl.go [hunk 2]
```diff
@@ -1406,6 +1423,48 @@ func (p *TerramateParser) parseTerramateSchema() (Config, error) {
...
+			panic("parsed terramate block not found")
+		}
+
+		errs.Append(
+			errors.E(ErrConfigConflict, stackblock.TypeRange,
+				"`stack` block conflicts with `terramate` block defined at %s",
+				tmblock.RawOrigins[0].TypeRange.String(),
+			),
+
+			errors.E(ErrConfigConflict, tmblock.RawOrigins[0].TypeRange,
```
Comment (katcipis) [inline hcl/hcl.go:1452]:
```suggestion
			),
			errors.E(ErrConfigConflict, tmblock.RawOrigins[0].TypeRange,
```

File: hcl/hcl_stack_test.go [hunk 3]
```diff
@@ -330,7 +352,8 @@ func TestHCLParserStack(t *testing.T) {
 			},
 		},
 		{
-			name: "after: empty set works",
+			name:      "after: empty set works",
```
Comment (katcipis) [inline hcl/hcl_stack_test.go:355]:
```suggestion
			name: "after: empty set works",
```

File: hcl/hcl_stack_test.go [hunk 4]
```diff
@@ -160,7 +181,8 @@ func TestHCLParserStack(t *testing.T) {
 			},
 		},
 		{
-			name: "empty name",
+			name:      "empty name",
```
Comment (katcipis) [inline hcl/hcl_stack_test.go:184]:
```suggestion
			name: "empty name",
```

I think the code is not formatted no ? :thinking:

**Review by katcipis — DISMISSED** [review]

**Comment by i4ki** [comment]
@katcipis I changed to report an error if `terramate` is not at the rootdir because my previous assumption was not valid, the docs still allow `stack + terramate` in the same dir **if it's root dir**.
The change broke some tests because `cfg.Terramate` was the easiest one to test the merging at the config level. I removed those testcases because we still have lots of tests for globals + import covering the same merge thing, so I think this is fine.

File: hcl/hcl.go [hunk 5]
```diff
@@ -1406,6 +1423,39 @@ func (p *TerramateParser) parseTerramateSchema() (Config, error) {
 	return config, nil
 }
 
+func (p *TerramateParser) checkConfigSanity(cfg Config) error {
+	logger := log.With().
+		Str("action", "TerramateParser.checkConfigSanity()").
```
Comment (katcipis) [inline hcl/hcl.go:1428]: Noice

**Review by katcipis — APPROVED** [review]

**Pull request merged**